plugin/core/views.py

```diff
@@ -781,28 +783,49 @@ def run(self, view_id: int, command: str, args: Optional[Dict[str, Any]] = None)
 
 
 COLOR_BOX_HTML = """
-<style>html {{padding: 0; background-color: transparent}}</style>
+<style>
+    html {{
+        padding: 0;
+        background-color: transparent;
+    }}
+    a {{
+        display: inline-block;
+        height: 0.8rem;
+        width: 0.8rem;
+        margin-top: 0.1em;
+        border: 1px solid color(var(--foreground) alpha(0.25));
+        background-color: {color};
+    }}
+</style>
 <body id='lsp-color-box'>
-<div style='padding: 0.4em;
-            margin-top: 0.2em;
-            border: 1px solid color(var(--foreground) alpha(0.25));
-            background-color: rgba({}, {}, {}, {})'>
-</div>
+    <a href='{command}'>&nbsp;</a>
 </body>"""
 
 
-def lsp_color_to_html(color_info: Dict[str, Any]) -> str:
-    color = color_info['color']
-    red = color['red'] * 255
-    green = color['green'] * 255
-    blue = color['blue'] * 255
-    alpha = color['alpha']
-    return COLOR_BOX_HTML.format(red, green, blue, alpha)
+def color_to_hex(color: Color) -> str:
+    red = int(color['red'] * 255)
+    green = int(color['green'] * 255)
+    blue = int(color['blue'] * 255)
+    alpha_dec = color['alpha']
+    if alpha_dec < 1:
+        return "#{:02x}{:02x}{:02x}{:02x}".format(red, green, blue, int(alpha_dec * 255))
+    return "#{:02x}{:02x}{:02x}".format(red, green, blue)
 
 
-def lsp_color_to_phantom(view: sublime.View, color_info: Dict[str, Any]) -> sublime.Phantom:
+def lsp_color_to_html(view: sublime.View, color_info: ColorInformation) -> str:
+    command = sublime.command_url('lsp_color_presentation', {
+        'params': {
+            'textDocument': text_document_identifier(view),
+            'color': color_info['color'],
```

Comment:
Have you though about passing the hole color_information dict,
instead of passing color and range dicts?

Reply:
Changed in https://github.com/sublimelsp/LSP/pull/2065/commits/78151f2792c4778385b554a5423feefb8a894855.

Not sure if you were thinking about doing it exactly this way, but it is actually a good idea to postpone the `text_document_identifier` function call to later when the color box is clicked.

I still need to use a new variable and extract the "color" and "range" values from ColorInformation one-by-one, because iirc otherwise pyright would complain about invalid assignment of a new unknown "textDocument" key for the params needed for the request.